solidity/contracts/middleware/InterchainQueryRouter.sol

```diff
@@ -140,27 +97,19 @@ contract InterchainQueryRouter is Router, IInterchainQueryRouter {
         bytes32, // router sender
         bytes calldata _message
     ) internal override {
-        Action action = Action(uint8(bytes1(_message[31])));
-        if (action == Action.DISPATCH) {
-            (
-                ,
-                address sender,
-                CallLib.Call[] memory calls,
-                bytes[] memory callbacks
-            ) = abi.decode(
-                    _message,
-                    (Action, address, CallLib.Call[], bytes[])
-                );
-            callbacks = calls._multicallAndResolve(callbacks);
-            _dispatch(_origin, abi.encode(Action.RESOLVE, sender, callbacks));
+        InterchainCallMessage.CallType calltype = _message.calltype();
+        bytes32 sender = _message.sender();
+        if (calltype == InterchainCallMessage.CallType.CALLBACK) {
```

Comment:
This is kind of confusing to me, since these are actually the static calls, and the *next* block (raw) is the callbacks.

Reply:
A comment could help clarify.